provisioner/node_pools.go

```diff
@@ -343,3 +456,43 @@ func nodePoolStackToNodePool(stack *cloudformation.Stack) *api.NodePool {
 	}
 	return nodePool
 }
+
+func KarpenterNodePoolConfigGetter(kubeClient *kubernetes.ClientsCollection) updatestrategy.NodePoolConfigGetter {
```

Comment:
It would be good practice to pass a `ctx` here and then use that in the client calls below where it currently uses `context.TODO()`.

Reply:
_correction_: It would be the `updatestrategy.NodePoolConfigGetter` that should accept a `context` as the first argument.

Reply:
will do

Reply:
done